consensus/polybft/state_sync_manager.go

```diff
@@ -0,0 +1,469 @@
+package polybft
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path"
+	"sync"
+
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/bitmap"
+	polybftProto "github.com/0xPolygon/polygon-edge/consensus/polybft/proto"
+	bls "github.com/0xPolygon/polygon-edge/consensus/polybft/signer"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/wallet"
+	"github.com/0xPolygon/polygon-edge/tracker"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/umbracle/ethgo"
+	"google.golang.org/protobuf/proto"
+)
+
+// stateSyncConfig holds the configuration data of state sync manager
+type stateSyncConfig struct {
+	stateSenderAddr types.Address
+	jsonrpcAddr     string
+	dataDir         string
+	topic           topic
+	key             *wallet.Key
+}
+
+// StateSyncManager is a struct that manages the workflow of
+// saving and querying state sync events, and creating, and submitting new commitments
+type StateSyncManager struct {
+	// configuration fields
+	logger hclog.Logger
+	state  *State
+
+	config *stateSyncConfig
+
+	// per epoch fields
+	lock         sync.Mutex
+	commitment   *Commitment
+	epoch        uint64
+	validatorSet *validatorSet
+}
+
+// topic is an interface for a gossip p2p message
+type topic interface {
+	Publish(obj proto.Message) error
+	Subscribe(handler func(obj interface{}, from peer.ID)) error
+}
+
+// NewStateSyncManager creates a new instance of state sync manager
+func NewStateSyncManager(logger hclog.Logger, state *State, config *stateSyncConfig) (*StateSyncManager, error) {
+	s := &StateSyncManager{
+		logger: logger.Named("state-sync"),
+		state:  state,
+		config: config,
+		lock:   sync.Mutex{},
+	}
+
+	return s, nil
+}
+
+// init subscribes to bridge topics (getting votes) and start the event tracker routine
+func (s *StateSyncManager) init() error {
+	if err := s.initTracker(); err != nil {
+		return err
+	}
+
+	if err := s.initTransport(); err != nil {
+		return err
```

Comment:
Add more context to error message, like for example:
```suggestion
		return fmt.Errorf("failed to initialize state sync transport layer", "error", err)
```

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1067/commits/acaaa3d20726fc29d0361e1b8472d936ceba4e67